src/main/kotlin/org/openrewrite/kotlin/internal/KotlinParserVisitor.kt

```diff
@@ -1341,99 +1341,78 @@ class KotlinParserVisitor(
     }
 
     private fun mapFunctionalCallArguments(firExpressions: List<FirExpression>): JContainer<Expression> {
+        val flattenedExpressions = firExpressions.stream()
+            .map { e: FirExpression ->
+                if (e is FirVarargArgumentsExpression) e.arguments else listOf(e)
+            }
+            .flatMap { it.stream() }
+            .collect(Collectors.toList())
+        val argumentCount = flattenedExpressions.size
+        val expressions: MutableList<JRightPadded<Expression>> = ArrayList(flattenedExpressions.size)
+        val isLastArgumentLambda = flattenedExpressions.isNotEmpty() && flattenedExpressions[argumentCount - 1] is FirLambdaArgumentExpression
+
+        var markers = Markers.EMPTY
         val args: JContainer<Expression>
-        if (firExpressions.size == 1) {
-            val firExpression = firExpressions[0]
-            args = if (firExpression is FirVarargArgumentsExpression) {
-                val argumentsExpression = firExpressions[0] as FirVarargArgumentsExpression
+        var saveCursor = cursor
+        val containerPrefix = whitespace()
+        var parenOrBrace = source[cursor++]
+        val saveCursor2 = cursor
+        val before = whitespace()
```

Comment:
I expect the tests' changes were unnecessary, and I suspect the issue is that the whitespace in `before` is not preserved. Should the `before` be added to a `J.Empty`?

Reply:
The changes to the existing test were unnecessary yes (just did them to fix a reported error). But the new test `trailingLambdaArgumentWithParentheses()` requires the changes made here.

Not sure if adding a `J.Empty` wouldn't break things, since there will also be another argument for the trailing lambda. But one thing which isn't yet covered (for which I assume we would want to add a dedicated marker) is if there is whitespace between the opening and closing empty parenthesis characters.